File: app/src/main/java/cn/chenyuanming/gankmeizhi/activity/BaseActivity.java
package cn.chenyuanming.gankmeizhi.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.jude.swipbackhelper.SwipeBackHelper;
import com.umeng.analytics.MobclickAgent;

/**
 * Created by Chen Yuanming on 2016/2/1.
 */
public abstract class BaseActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SwipeBackHelper.onCreate(this);
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        SwipeBackHelper.onPostCreate(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        SwipeBackHelper.onDestroy(this);
    }

    public void onResume() {
        super.onResume();
        MobclickAgent.onPageStart(getClass().getSimpleName());
        MobclickAgent.onResume(this);
    }

    public void onPause() {
        super.onPause();
        MobclickAgent.onPageEnd(getClass().getSimpleName());
        MobclickAgent.onPause(this);
    }
}

File: app/src/main/java/cn/chenyuanming/gankmeizhi/activity/BaseActivity.kt
package cn.chenyuanming.gankmeizhi.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

import com.jude.swipbackhelper.SwipeBackHelper
import com.umeng.analytics.MobclickAgent

/**
 * Created by Chen Yuanming on 2016/2/1.
 */
abstract class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        SwipeBackHelper.onCreate(this)
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        SwipeBackHelper.onPostCreate(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        SwipeBackHelper.onDestroy(this)
    }

    public override fun onResume() {
        super.onResume()
        MobclickAgent.onPageStart(javaClass.simpleName)
        MobclickAgent.onResume(this)
    }

    public override fun onPause() {
        super.onPause()
        MobclickAgent.onPageEnd(javaClass.simpleName)
        MobclickAgent.onPause(this)
    }
}
